### ISSUE TITLE
Suggestion: Avoid sudo inside installer script

### ISSUE THREAD

**Issue by shadmansaleh** [ChristianChiarulli/LunarVim#1044]
There are 22 sudo statements in installer script .

I think any script elevating it's permission without users conscience is a terrible idea . And that one being directly executed while being fetched over network is even horrifying .

I feel like lunarvim installer shouldn't install dependencies directly. Also installing lvim script on ~/.local/bin is probably a better idea then /usr/local .

Maintaining ways how system packages are installed across different destributions is unnecessary task . You can just ask user to install the dependecies .

If the script needs root access it should depend on user to provide it instead of elevating itself. That way at lest the user knows it's being executed as root . like for example you need to do `sudo make install` . It's easy to add sudo to install cammands . But user is required to provide it .

Also bash < curl is a bad practice from security stand point . It's already a bad idea toe execute anything directly off web . But can cause trouble if for some reason netwrok interuption can cause partial script to be executed . And that as root can cause a lot of damage .

**Comment by Uzaaft** [comment]
While I understand the reaction, the best way to solve this is to go ahead and rewrite the bash script to fix the issues you have with the install script. 
IMHO, the way you write the issue makes me less inclined to help you, if this was an issue I could solve.

**Comment by shadmansaleh** [comment]
Ok sorry the language was to harsh. You see I ran the script then it got stuck (I guess PackerSync) . Then opened the script was just too shocked to see the sudos :D

**Comment by abzcoding** [comment]
> Sorry that was a bit too harsh.
> 
> You see I ran the script, it got stuck after a while (I guess PackerSync) . Then opened the script and was just too shocked to see all the sudos there

😅 why so serious 🃏
atm there are tons of PRs that want to try and change the installer script (#648 #651 #766 )
but we are trying to make Lunarvim a separate pkg so instead of using the installer you can install it using the package installer of your choice:
```bash
sudo apt-get install lunarvim
brew install lunarvim
sudo pacman -S lunarvim
# you get the point
```
so stay tuned

**Comment by abzcoding** [comment]
Gonna be fixed by #1052
